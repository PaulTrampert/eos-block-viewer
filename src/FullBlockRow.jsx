import React from 'react';
import PropTypes from 'prop-types'

const FullBlockRow = ({block, onClick}) => (
  <tr onClick={onClick}>
    <td colSpan={3}>
      <pre>
        {JSON.stringify(block, null, 2)}
      </pre>
    </td>
  </tr>
)

FullBlockRow.propTypes = {
  block: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

export default FullBlockRow;