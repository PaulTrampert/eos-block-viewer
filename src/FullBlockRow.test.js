import React from 'react';
import {shallow} from 'enzyme';
import FullBlockRow from './FullBlockRow';
import { Tabs } from 'react-bootstrap';

describe('FullBlockRow', () => {
  let subject;
  let block;
  let onClick;

  beforeEach(() => {
    block = {
      id: 'asdfasdf',
      timestamp: '2020-02-08T20:20:20Z',
      transactions: []
    };

    onClick = jest.fn();

    subject = shallow(<FullBlockRow block={block} onClick={onClick} />);
  });

  describe('setTabKey', () => {
    let event;
    beforeEach(() => {
      event = {
        stopPropagation: jest.fn()
      };
    });

    it('stops the event propagation', () => {
      subject.instance().setTabKey("key", event);

      expect(event.stopPropagation).toHaveBeenCalled();
    });

    it('sets the tabKey in state', () => {
      subject.instance().setTabKey("key", event);

      expect(subject.state().tabKey).toBe('key');
    });
  });

  describe('render', () => {
    it('renders tabs for the raw contents of the block and the ricardian contracts', () => {
      expect(subject).toMatchSnapshot();
    });

    it('binds the onClick event', () => {
      expect(subject.find('tr').props().onClick).toBe(onClick);
    });

    it('binds the onSelect event of tabs', () => {
      expect(subject.find(Tabs).props().onSelect).toBe(subject.instance().setTabKey);
    });

    describe('when tabKey is "raw"', () => {
      it('does not render the RicardianBlockView and only renders the RawBlockView', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe('when tabKey is "ricardian"', () => {
      beforeEach(() => {
        subject.setState({tabKey:'ricardian'});
      });

      it('does not render the RawBlockView and only renders the RicardianBlockView', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});