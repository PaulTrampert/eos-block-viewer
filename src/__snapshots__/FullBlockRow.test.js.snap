// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullBlockRow render renders tabs for the raw contents of the block and the ricardian contracts 1`] = `
<tr
  onClick={[MockFunction]}
>
  <td
    colSpan={3}
  >
    <Tabs
      activeKey="raw"
      mountOnEnter={false}
      onSelect={[Function]}
      unmountOnExit={false}
      variant="tabs"
    >
      <Tab
        eventKey="raw"
        title="Raw View"
      >
        <RawBlockView
          block={
            Object {
              "id": "asdfasdf",
              "timestamp": "2020-02-08T20:20:20Z",
              "transactions": Array [],
            }
          }
        />
      </Tab>
      <Tab
        eventKey="ricardian"
        title="Ricardian Contracts"
      />
    </Tabs>
  </td>
</tr>
`;

exports[`FullBlockRow render when tabKey is "raw" does not render the RicardianBlockView and only renders the RawBlockView 1`] = `
<tr
  onClick={[MockFunction]}
>
  <td
    colSpan={3}
  >
    <Tabs
      activeKey="raw"
      mountOnEnter={false}
      onSelect={[Function]}
      unmountOnExit={false}
      variant="tabs"
    >
      <Tab
        eventKey="raw"
        title="Raw View"
      >
        <RawBlockView
          block={
            Object {
              "id": "asdfasdf",
              "timestamp": "2020-02-08T20:20:20Z",
              "transactions": Array [],
            }
          }
        />
      </Tab>
      <Tab
        eventKey="ricardian"
        title="Ricardian Contracts"
      />
    </Tabs>
  </td>
</tr>
`;

exports[`FullBlockRow render when tabKey is "ricardian" does not render the RawBlockView and only renders the RicardianBlockView 1`] = `
<tr
  onClick={[MockFunction]}
>
  <td
    colSpan={3}
  >
    <Tabs
      activeKey="ricardian"
      mountOnEnter={false}
      onSelect={[Function]}
      unmountOnExit={false}
      variant="tabs"
    >
      <Tab
        eventKey="raw"
        title="Raw View"
      />
      <Tab
        eventKey="ricardian"
        title="Ricardian Contracts"
      >
        <RicardianBlockView
          block={
            Object {
              "id": "asdfasdf",
              "timestamp": "2020-02-08T20:20:20Z",
              "transactions": Array [],
            }
          }
        />
      </Tab>
    </Tabs>
  </td>
</tr>
`;
